mailbox/api/src/main/java/org/apache/james/mailbox/events/MailboxListener.java

```diff
@@ -77,7 +82,17 @@ default boolean isHandling(Event event) {
      *
      * @param event not null
      */
-    void event(Event event) throws Exception;
+    default void event(Event event) throws Exception {
+        Mono.from(eventReactive(event))
+            .subscribeOn(Schedulers.elastic())
+            .block();
+    }
+
+    default Publisher<Void> eventReactive(Event event) {
```

Comment:
if you don't implement event or eventReactive, you have an infinite loop

Reply:
I know